vortex-array/src/arrays/constant/compute/take.rs

```diff
@@ -0,0 +1,66 @@
+use vortex_error::VortexResult;
+use vortex_mask::AllOr;
+use vortex_scalar::Scalar;
+
+use crate::arrays::{ConstantArray, ConstantEncoding};
+use crate::builders::{ArrayBuilderExt, builder_with_capacity};
+use crate::compute::TakeFn;
+use crate::{Array, ArrayRef};
+
+impl TakeFn<&ConstantArray> for ConstantEncoding {
+    fn take(&self, array: &ConstantArray, indices: &dyn Array) -> VortexResult<ArrayRef> {
+        match indices.validity_mask()?.boolean_buffer() {
+            AllOr::All => {
+                Ok(ConstantArray::new(array.scalar().clone(), indices.len()).into_array())
+            }
+            AllOr::None => Ok(ConstantArray::new(
+                Scalar::null(array.dtype().clone()),
+                indices.len(),
+            )
+            .into_array()),
+            AllOr::Some(v) => {
+                let mut result_builder =
```

Comment:
Is this not just a sparse array? Fill value / patches can model valid/invalid, depending on how many nulls there are?

Reply:
I realise we'd have to bring sparse back into vortex-array

Reply:
You’re correct it’s just a sparse array and depending on the density of indices you can switch whether null or value is the fill

Reply:
The builder approach is also fine, but we probably want to append_array, and then set_validity? That should be doable on an empty length builder in basically constant time

Reply:
I have added set_validity option to our builders